src/ray/core_worker/future_resolver.h

```diff
@@ -0,0 +1,68 @@
+#ifndef RAY_CORE_WORKER_FUTURE_RESOLVER_H
+#define RAY_CORE_WORKER_FUTURE_RESOLVER_H
+
+#include <memory>
+
+#include "ray/common/id.h"
+#include "ray/core_worker/store_provider/memory_store/memory_store.h"
+#include "ray/protobuf/core_worker.pb.h"
+#include "ray/rpc/worker/core_worker_client.h"
+
+namespace ray {
+
+/// Max time between requests to the owner to check whether the object has been
+/// evicted.
```

Comment:
It would be more clear to say "whether the object is still being computed", since that's what is actually going on. You're not only detecting eviction but other scenarios such as failure.